packages/template-retail-react-app/app/components/product-tile/index.jsx

```diff
@@ -56,54 +65,135 @@ export const Skeleton = () => {
     // RecommendedProducts provides a localized product name as `name` and non-localized product
     // name as `productName`. ProductList provides a localized name as `productName` and does not
     // use the `name` property.
     const localizedProductName = product.name ?? product.productName
 
-    const {currency: activeCurrency} = useCurrency()
-    const isFavouriteLoading = useRef(false)
-    const styles = useMultiStyleConfig('ProductTile')
-
     return (
         <Box {...styles.container}>
-            <Link
-                data-testid="product-tile"
-                to={productUrlBuilder({id: productId}, intl.local)}
-                {...styles.link}
-                {...rest}
-            >
+            <Link data-testid="product-tile" to={productUrl} {...styles.link} {...rest}>
                 <Box {...styles.imageWrapper}>
-                    {image && (
-                        <AspectRatio {...styles.image}>
-                            <DynamicImage
-                                src={`${image.disBaseLink || image.link}[?sw={width}&q=60]`}
-                                widths={dynamicImageProps?.widths}
-                                imageProps={{
-                                    // treat img as a decorative item, we don't need to pass `image.alt`
-                                    // since it is the same as product name
-                                    // which can cause confusion for individuals who uses screen readers
-                                    alt: '',
-                                    ...dynamicImageProps?.imageProps
-                                }}
-                            />
-                        </AspectRatio>
-                    )}
+                    <AspectRatio {...styles.image}>
+                        <DynamicImage
+                            src={`${
+                                image?.disBaseLink ||
+                                image?.link ||
+                                product?.image?.disBaseLink ||
+                                product?.image?.disBaseLink
+                            }[?sw={width}&q=60]`}
+                            widths={dynamicImageProps?.widths}
+                            imageProps={{
+                                // treat img as a decorative item, we don't need to pass `image.alt`
+                                // since it is the same as product name
+                                // which can cause confusion for individuals who uses screen readers
+                                alt: '',
+                                ...dynamicImageProps?.imageProps
+                            }}
+                        />
+                    </AspectRatio>
                 </Box>
 
+                {/* Swatches */}
+                {variationAttributes
+                    ?.filter(({id}) => selectableAttributeId === id)
+                    ?.map(({id, values}) => (
+                        <SwatchGroup
```

Comment:
Nit: We should make sure the attribute name (color/size) is included in the label for swatch. This is because not all color names are easy to regconize (e.g like seagrass or Light Seabreeze), without the context as colors, it may be hard for screen reader users to know what it is actually about.

Reply:
Good catch, it looks like we were overloading the "label" prop to be used for both the label, and the aria-label. As I didn't need or want the label to be displayed in this scenario I didn't provide a the label to the swatch group. This as expected stopped the label from showing, but also meant there was no aria-label. 

So resolve this I've added an explicit "ariaLabel" prop that will take precedence over the label when passed in. This also won't effect the showing or hiding of the visual label.

Reply:
@bendvc the name here is still just the color value. The screen reader will says something like this "Seagrass, selected, radiogroup, 1 out of 2". It still misses the context of attribution type. I think it will be more helpful to have it like this so the screen will read out loud "Color Seagrass, select, ..." 
```
ariaLabel={`${selectableAttributeId}-${name}`}
```